File: shared/commonResources/src/commonMain/kotlin/pl/msiwak/multiplatform/commonResources/theme/Theme.kt
package pl.msiwak.multiplatform.commonResources.theme

import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import dev.icerock.moko.resources.compose.colorResource
import pl.msiwak.multiplatform.commonResources.SR

@Composable
private fun appColorScheme() = lightColorScheme(
    primary = colorResource(SR.colors.colorPrimary),
    onPrimary = colorResource(SR.colors.colorOnPrimary),
    secondary = colorResource(SR.colors.colorSecondary),
    onSecondary = colorResource(SR.colors.colorOnSecondary),
    tertiary = colorResource(SR.colors.colorTertiary),
    onTertiary = colorResource(SR.colors.colorOnTertiary),
    surface = colorResource(SR.colors.colorSurface),
    onSurface = colorResource(SR.colors.colorOnSurface),
    background = colorResource(SR.colors.background),
    error = colorResource(SR.colors.colorError),
    onError = colorResource(SR.colors.colorOnError)
)

@Composable
fun AppTheme(
    content: @Composable () -> Unit
) {
    val colorScheme = appColorScheme()

    CompositionLocalProvider(LocalOwnColorScheme provides OwnColorScheme(true)) {
        MaterialTheme(
            colorScheme = colorScheme,
            typography = Typography,
            content = content
        )
    }
}

File: multiplatform/iosApp/iosApp/Theme.swift
import shared
import SwiftUI

extension SwiftUI.Color {
    static let colorPrimary = Color.p(SR.colors().colorPrimary.getUIColor())
    static let onPrimary = Color(SR.colors().colorOnPrimary.getUIColor())
    static let colorSecondary = Color(SR.colors().colorSecondary.getUIColor())
    static let onSecondary = Color(SR.colors().colorOnSecondary.getUIColor())
    static let colorTertiary = Color(SR.colors().colorTertiary.getUIColor())
    static let onTertiary = Color(SR.colors().colorOnTertiary.getUIColor())
    static let surface = Color(SR.colors().colorSurface.getUIColor())
    static let onSurface = Color(SR.colors().colorOnSurface.getUIColor())
    static let background = Color(SR.colors().background.getUIColor())
    static let error = Color(SR.colors().colorError.getUIColor())
    static let onError = Color(SR.colors().colorOnError.getUIColor())
    
    static let gray = Color(SR.colors().gray.getUIColor())
}
